### PR TITLE
EUID marketplace offering

### PR DESCRIPTION
**Pull request by cody-constine-ttd**

### CODE REVIEW

File: scripts/aws/Dockerfile [hunk 1]
```diff
@@ -4,9 +4,11 @@ WORKDIR /app
 ARG JAR_NAME=uid2-operator
 ARG JAR_VERSION=1.0.0
 ARG IMAGE_VERSION=1.0.0.unknownhash
+ARG REGION=UID2
```
Comment (atarassov-ttd) [inline scripts/aws/Dockerfile:7]: Should this be renamed to `SCOPE` or `IDENTITY_SCOPE` to match the setting name in the config file?

File: scripts/aws/config.json [hunk 2]
```diff
@@ -1 +1,14 @@
-{ "service_instances": 4, "clients_metadata_path": "https://core-prod.uidapi.com/clients/refresh", "keys_metadata_path": "https://core-prod.uidapi.com/key/refresh", "keys_acl_metadata_path": "https://core-prod.uidapi.com/key/acl/refresh", "salts_metadata_path": "https://core-prod.uidapi.com/salt/refresh", "optout_metadata_path": "https://optout-prod.uidapi.com/optout/refresh", "core_attest_url": "https://core-prod.uidapi.com/attest", "core_api_token": "your-api-key", "optout_s3_path_compat": true, "optout_api_uri": "https://core-prod.uidapi.com/optout/replicate", "optout_api_token": "your-api-key", "enclave_platform": "aws-nitro" }
```
Comment (atarassov-ttd) [inline scripts/aws/config.json:1]: What is this file used for?

File: scripts/aws/conf/integ-config.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (atarassov-ttd) [inline scripts/aws/conf/integ-config.json:1]: Should this file be named `integ-uid2-config.json` for consistency?

File: Makefile.nitro [hunk 4]
```diff
@@ -45,9 +49,20 @@ build/load_config.py: ./scripts/aws/load_config.py
 build/make_config.py: ./scripts/aws/make_config.py
 	cp ./scripts/aws/make_config.py ./build/
 
+build/configs: build/conf/prod-config.json build/conf/integ-config.json build/conf/prod-euid-config.json build/conf/integ-euid-config.json
+
 build/conf/prod-config.json: build_artifacts ./scripts/aws/conf/prod-config.json
```
Comment (atarassov-ttd) [inline Makefile.nitro:54]: Should this file be renamed to `prod-uid2-config.json` for consistency?

File: scripts/aws/make_config.py [hunk 5]
```diff
@@ -14,8 +14,9 @@ def apply_override(config, overrides, key, type):
 
 
 config_path = sys.argv[1]
-overrides_path = sys.argv[2]
-thread_count = int(sys.argv[3])
+integ_config_path = sys.argv[2]
+overrides_path = sys.argv[3]
+thread_count = int(sys.argv[4])
 
 config = load_json(config_path)
```
Comment (atarassov-ttd) [inline scripts/aws/make_config.py:21]: Rather than have overrides section for integ below, have you considered doing something like this:

```
overrides = load_json(overrides_path)
config = load_json(integ_config_path if override.get('environment') == 'integ' else prod_config_path)
```

File: scripts/aws/config.json (hunk 2 above)
Comment (cody-constine-ttd) [reply scripts/aws/config.json:1]: I think was just used for testing

File: scripts/aws/make_config.py (hunk 5 above)
Comment (cody-constine-ttd) [reply scripts/aws/make_config.py:21]: I could be convinced but this way makes all the other settings besides the URLs easier to keep in sync

File: scripts/aws/EUID_CloudFormation.template.yml [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+          - VpcSubnet1
+          - VpcSubnet2
+          - NewVpcCidr
+          - Subnet1Cidr
+          - Subnet2Cidr
+    ParameterLabels:
+      APIToken:
+        default: OPERATOR_KEY provided by EUID Administrator
+      DeployToEnvironment:
+        default: EUID environment to deploy to. Prod - production; Integ - integration test.
```
Comment (atarassov-ttd) [inline scripts/aws/EUID_CloudFormation.template.yml:116]: Should `prod` and `integ` be lowercase here?

File: scripts/aws/EUID_CloudFormation.template.yml [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+        default: If choose to create new VPC, enter CIDR for new VPC (required in case of new VPC)
+      CustomizeEnclaceResource:
+        default: Enclave resouce configuration auto calculated or manual
+      EnclavememoryinMB:
+        default: If choose to false for CustomizeEnclaceResource, enter memory for Enclave in MB
+      EnclaveCPUCount:
+        default: If choose to false for CustomizeEnclaceResource, enter CPU count for Enclave
+Mappings:
+  RegionMap:
+    us-east-1:
```
Comment (atarassov-ttd) [inline scripts/aws/EUID_CloudFormation.template.yml:141]: Should this list be reduced to only EU locations?

File: scripts/aws/Dockerfile [hunk 8]
```diff
@@ -4,9 +4,11 @@ WORKDIR /app
 ARG JAR_NAME=uid2-operator
 ARG JAR_VERSION=1.0.0
 ARG IMAGE_VERSION=1.0.0.unknownhash
+ARG IDENTITY_SCOPE=UID2
```
Comment (atarassov-ttd) [inline scripts/aws/Dockerfile:7]: Should the default value be removed here?

File: scripts/aws/entrypoint.sh [hunk 9]
```diff
@@ -24,7 +24,12 @@ echo "AWS_REGION_NAME=$AWS_REGION_NAME"
 echo "127.0.0.1 secretsmanager.$AWS_REGION_NAME.amazonaws.com" >> /etc/hosts
 
 python3 /app/load_config.py >/app/conf/config-overrides.json
-python3 /app/make_config.py /app/conf/config.json /app/conf/config-overrides.json $(nproc) >/app/conf/config-final.json
+
+if [ "$IDENTITY_SCOPE" = 'UID2' ]; then
+  python3 /app/make_config.py /app/conf/prod-uid2-config.json /app/conf/integ-uid2-config.json /app/conf/config-overrides.json $(nproc) >/app/conf/config-final.json
+elif [ "$IDENTITY_SCOPE" = 'EUID' ]; then
+  python3 /app/make_config.py /app/conf/prod-euid-config.json /app/conf/integ-euid-config.json /app/conf/config-overrides.json $(nproc) >/app/conf/config-final.json
+fi
```
Comment (atarassov-ttd) [inline scripts/aws/entrypoint.sh:32]: Add `else` block and explicitly fail with an error message?

File: scripts/aws/entrypoint.sh [hunk 10]
```diff
@@ -11,7 +11,7 @@ ifconfig lo 127.0.0.1
 /app/vsockpx --config /app/proxies.nitro.yaml --daemon --workers $(( $(nproc) * 2 )) --log-level 3
 
 # -- load config via proxy
-export UID2_CONFIG_SECRET_KEY=$([[ "$(curl -s -x socks5h://127.0.0.1:3305 http://169.254.169.254/latest/user-data | grep UID2_CONFIG_SECRET_KEY=)" =~ ^export\ UID2_CONFIG_SECRET_KEY=\"(.*)\"$ ]] && echo ${BASH_REMATCH[1]} || echo "uid2-operator-config-key")
+export UID2_CONFIG_SECRET_KEY=$([[ "$(curl -s -x socks5h://127.0.0.1:3305 http://169.254.169.254/latest/user-data | grep UID2_CONFIG_SECRET_KEY=)" =~ ^export\ UID2_CONFIG_SECRET_KEY=\"(.*)\" ]] && echo ${BASH_REMATCH[1]} || echo "uid2-operator-config-key")
```
Comment (atarassov-ttd) [inline scripts/aws/entrypoint.sh:14]: Why is this changed?

File: scripts/aws/make_config.py (hunk 5 above)
Comment (atarassov-ttd) [reply scripts/aws/make_config.py:21]: But then you still have two files with full set of configs which is very confusing. I see two options here:
1. Have prod config to contain all settings and have integ config only have the (overridden) paths.
2. Have a single base config without paths and then have prod/integ config only with paths.

I like the second approach more since in the first one if a new path is added to prod config, but not to integ, the operator would behave weirdly which would be more difficult to troubleshoot.

File: scripts/aws/entrypoint.sh (hunk 10 above)
Comment (cody-constine-ttd) [reply scripts/aws/entrypoint.sh:14]: The regex wasn't working with the `$` at the end

File: scripts/aws/Dockerfile (hunk 8 above)
Comment (cody-constine-ttd) [reply scripts/aws/Dockerfile:7]: It feels like UID2 should be the default value.

File: scripts/aws/config.json (hunk 2 above)
Comment (scong-ttd) [reply scripts/aws/config.json:1]: This used to be the config template that enclave consumes. Doesn't seem to be used anymore.

File: Makefile.nitro [hunk 11]
```diff
@@ -33,9 +33,13 @@ clean:
 	cd build/dante-1.4.3; ./configure
 	cd build/dante-1.4.3; make
 
-build_eif: uid2operator.eif
+build_eif: uid2operator.eif euidoperator.eif
```
Comment (scong-ttd) [inline Makefile.nitro:36]: Do you plan to send this euidoperator.eif onto the AMI and rename it to be uid2operator? Otherwise we need to `uid2operator.service`

File: scripts/aws/entrypoint.sh (hunk 10 above)
Comment (scong-ttd) [inline scripts/aws/entrypoint.sh:14]: This is not consuming `EUID_CONFIG_SECRET_KEY`
